### PR TITLE
Get the list of contributors for GitHub repositories

### PR DESCRIPTION
**Pull request by giordano**

### CODE REVIEW

File: src/AnalyzeRegistry.jl [hunk 1]
```diff
@@ -373,9 +389,19 @@ function analyze(dir::AbstractString; repo = "", reachable=true, subdir="")
         license_files = LicenseTableEltype[]
         lines_of_code = LoCTableEltype[]
     end
+
+    # If the repository is on GitHub and we have a non-anonymous GitHub
+    # authentication, get the list of contributors
+    contributors = if auth isa GitHub.AnonymousAuth || occursin("github.com", repo)
```
Comment (ericphanson) [inline src/AnalyzeRegistry.jl:395]:
```suggestion
    contributors = if !(auth isa GitHub.AnonymousAuth) || occursin("github.com", repo)
```

File: src/AnalyzeRegistry.jl [hunk 2]
```diff
@@ -36,6 +37,7 @@ struct Package
     license_files::Vector{LicenseTableEltype} # a table of all possible license files
     licenses_in_project::Vector{String} # any licenses in the `license` key of the Project.toml
     lines_of_code::Vector{LoCTableEltype}
+    contributors::Vector{String}
```
Comment (ericphanson) [inline src/AnalyzeRegistry.jl:40]: Add to the docs?

File: src/AnalyzeRegistry.jl [hunk 3]
```diff
@@ -373,9 +389,19 @@ function analyze(dir::AbstractString; repo = "", reachable=true, subdir="")
         license_files = LicenseTableEltype[]
         lines_of_code = LoCTableEltype[]
     end
+
+    # If the repository is on GitHub and we have a non-anonymous GitHub
+    # authentication, get the list of contributors
+    contributors = if auth isa GitHub.AnonymousAuth || occursin("github.com", repo)
+        repo_name = replace(replace(repo, r"^https://github\.com/" => ""), r"\.git$" => "")
+        String[c["contributor"].login for c in GitHub.contributors(GitHub.repo(repo_name; auth); auth)[1] if c["contributor"].login ∉ ("JuliaTagBot", "github-actions[bot]")]
```
Comment (ericphanson) [inline src/AnalyzeRegistry.jl:397]: Is zeptodoctor one we should exclude too? (If I’m remembering that name right...)

File: src/AnalyzeRegistry.jl (hunk 1 above)
Comment (giordano) [reply src/AnalyzeRegistry.jl:395]: Yeah, I don't know what I was doing here :smile:

File: src/AnalyzeRegistry.jl (hunk 2 above)
Comment (giordano) [reply src/AnalyzeRegistry.jl:40]: Yep, I still need to do it, thanks for the reminder!

**Comment by ericphanson** [comment]
Wow, didn’t see this is only 5 mins old aha! Looks great though

File: src/AnalyzeRegistry.jl (hunk 3 above)
Comment (giordano) [reply src/AnalyzeRegistry.jl:397]: Yeah, we need to collect all bots we know, these are the first two I found.

File: src/AnalyzeRegistry.jl [hunk 4]
```diff
@@ -334,7 +350,7 @@ Package DataFrames:
 
 ```
 """
-function analyze(dir::AbstractString; repo = "", reachable=true, subdir="")
+function analyze(dir::AbstractString; repo = "", reachable=true, subdir="", auth::GitHub.Authorization=github_auth())
```
Comment (giordano) [inline src/AnalyzeRegistry.jl:353]: We may want to add the `auth` keyword argument to all functions that eventually call this one.

File: src/AnalyzeRegistry.jl (hunk 2 above)
Comment (giordano) [reply src/AnalyzeRegistry.jl]: Done

File: src/AnalyzeRegistry.jl (hunk 4 above)
Comment (giordano) [reply src/AnalyzeRegistry.jl:394]: Done

**Comment by giordano** [comment]
I've made `contributors` a `Dict{String,Int}(contributor => contributions)`, which should allow us to collect more interesting statistics.  I've also added a section to the documentation showing an example of use: https://giordano.github.io/AnalyzeRegistry.jl/previews/PR32/#Contributors-to-the-repository.  Since you seemed happy about this, I'm going to merge this when CI is green, but feel free to open issues/pull requests if you see any possible fix/improvement!

**Pull request merged**